subtensor-api/src/subtensorapi/__init__.py

```diff
@@ -0,0 +1,327 @@
+        """
+        try:
+            with open(os.path.join(os.path.expanduser(json_file_location))) as f:
+                file_data = f.read()
+            return cls._load_neurons_from_blockAtRegistration_all_file_data(file_data)
+        except FileNotFoundError:
+            raise FastSyncFileException('{} not found. Try calling fast_sync_neurons() first.', json_file_location)
+        except OSError:
+            raise FastSyncFileException('Could not read {}', json_file_location)
+
+    @classmethod
+    def _load_neurons_from_blockAtRegistration_all_file_data(cls, file_data: str) -> List[int]:
+        """
+        Loads neurons from the blockAtRegistration_all JSON file data
+        
+        Raises: FastSyncFormatException if the file is not in the correct format
+
+        Returns: List[int]
+            a list of the blockAtRegistration numbers
+        """
+        try:
+            data = json.loads(file_data)
+        except json.JSONDecodeError:
+            raise FastSyncFormatException('Could not parse blockAtRegistration JSON file data as json')
+
+        # all the large ints are strings
+        if not isinstance(data, list):
+            raise FastSyncFormatException('Expected a JSON array at the top level')
+        
+        try:
+            # validate the blockAtRegistration data
+            blockAtRegistration_all: List[int] = [
+                int(blockAtRegistration) for blockAtRegistration in tqdm(data, "Parsing blockAtRegistration_all data")
+            ]
+        except Exception as e:
+            raise FastSyncFormatException('Could not parse blockAtRegistration JSON file data: {}'.format(e))
+            
+        return blockAtRegistration_all
+
+    @classmethod
+    def _load_neurons_from_metragraph_file_data(cls, file_data: str) -> List[SimpleNamespace]:
+        """
+        Loads neurons from the metagraph file data
+        See: https://github.com/opentensor/subtensor-node-api#neuron-structure
+        
+        Raises: FastSyncFormatException if the file is not in the correct format
+
+        Returns: List[SimpleNamespace]
+            a list of the Neurons
+        """
+        try:
+            data = json.loads(file_data)
+        except json.JSONDecodeError:
+            raise FastSyncFormatException('Could not parse metagraph file data as json')
+
+        # all the large ints are strings
+        if not isinstance(data, list):
+            raise FastSyncFormatException('Expected a JSON array at the top level')
+        
+        neurons: List[SimpleNamespace] = []
+        try:
+            
+            for neuron_data in tqdm(data, "Parsing Neuron Data"):
+                # validate the neuron data
+                neuron = SimpleNamespace( **neuron_data )
+                # hotkey and coldkey are strings
+                # uid is an int
+                # active is an int
+                # ip is a string
+                # ip_type is an int
+                # port is an int
```

Comment:
Why do we have these comments? where are those fields?

Reply:
Those fields are added in the keyword deconstruction `neuron_data**`

I added the comments to make it clear that they're already parsed properly from the JSON

Reply:
mmm it wasn't that clear to me. Do you think we can improve that commenting that the line before the deconstruaction? We could mention that and adding all the files enumerating them, what do you think about that?